### ISSUE TITLE
KEGG search results should only display products from which the KEGG hit comes from

### ISSUE THREAD

**Issue by aclum** [microbiomedata/issues#169]
**Is your feature request related to a problem? Please describe.**
When searching by a KEGG search term it should only products where it makes sense to have KEGG results (metagenome, metatranscriptome, and proteomics) 
Example (https://data-dev.microbiomedata.org/?q=Ch8IABAAGAEiFyJLRUdHLk9SVEhPTE9HWTpLMDAwMDki)

**Describe the solution you'd like**
A metagenome sample has a hit to the KEGG term and also has a metatranscriptome with no KEGG hit and metabolomics the sample results should only show the metagenome product. The count of metagenomes at the study level should be the number of metagenomes in that study that have the KEGG term, not the total number of metagenome projects in the entire study. 
Minimum requirement for GSP would be to hid metabolomics and natural organic matter results for studies and samples

**Describe alternatives you've considered**
Minimum requirement for GSP would be to hid metabolomics and natural organic matter results for studies and samples

**Acceptance Criteria**
Search by KEGG term hides shows the number of projects within a study that have hits and only those with products. At the sample level only show products with the KEGG hit.
![Screenshot 2023-03-30 at 7 47 23 PM](https://user-images.githubusercontent.com/8196591/229011064-890d2550-e90f-4d68-8d0f-191b05092de1.png)

**Comment by jeffbaumes** [comment]
Yes, it appears that for KEGG search, the chips represent the data for all samples in the full study, even if the sample results below have a subset of samples, and if the bar chart of omics types represents only metaG, metaP, metaT data. It looks like all that is filtered in the study list that a study is not listed if there is no KEGG hit at all from that study, otherwise the full study is represented.

We should work to add this functionality, but this feels like a larger fix beyond GSP timeframe as it would touch deeper faceted query functionality that is not trivial. I believe @naglepuff should work on the other minor issues first and we can see where we are at that point to work on this one.

**Comment by jeffbaumes** [comment]
We need to decide more explicitly what this solution will look like. A few questions: Does the user not want to know that there is associated NOM data for samples where the KEGG term matched? What about the sample results, should NOM data appear there or should it be removed?

**Comment by aclum** [comment]
@naglepuff are you good to work on this next sprint?
